dataframely/collection.py

```diff
@@ -620,84 +625,230 @@ def serialize(cls) -> str:
         members which are not provided in the current collection.
 
+        In addition, one JSON file named ``schema.json`` is written, serializing the
+        collection's definition for fast reads.
+
         Args:
             directory: The directory where the Parquet files should be written to. If
                 the directory does not exist, it is created automatically, including all
                 of its parents.
+            kwargs: Additional keyword arguments passed directly to
+                :meth:`polars.write_parquet` of all members. ``metadata`` may only be
+                provided if it is a dictionary.
+
+        Attention:
+            This method suffers from the same limitations as :meth:`Schema.serialize`.
         """
-        directory.mkdir(parents=True, exist_ok=True)
+        path = Path(directory) if isinstance(directory, str) else directory
+        path.mkdir(parents=True, exist_ok=True)
+
+        with open(path / "schema.json", "w") as f:
+            f.write(self.serialize())
+
+        member_schemas = self.member_schemas()
         for key, lf in self.to_dict().items():
-            lf.collect().write_parquet(directory / f"{key}.parquet")
+            member_schemas[key].write_parquet(
+                lf.collect(),  # type: ignore
+                path / f"{key}.parquet",
+                **kwargs,
+            )
+
+    def sink_parquet(self, directory: str | Path, **kwargs: Any) -> None:
+        """Stream the members of this collection into parquet files in a directory.
+
+        This method writes one parquet file per member into the provided directory.
+        Each parquet file is named ``<member>.parquet``. No file is written for optional
+        members which are not provided in the current collection.
+
+        In addition, one JSON file named ``schema.json`` is written, serializing the
+        collection's definition for fast reads.
+
+        Args:
+            directory: The directory where the Parquet files should be written to. If
+                the directory does not exist, it is created automatically, including all
+                of its parents.
+            kwargs: Additional keyword arguments passed directly to
+                :meth:`polars.sink_parquet` of all members. ``metadata`` may only be
+                provided if it is a dictionary.
+
+        Attention:
+            This method suffers from the same limitations as :meth:`Schema.serialize`.
+        """
+        path = Path(directory) if isinstance(directory, str) else directory
+        path.mkdir(parents=True, exist_ok=True)
+
+        with open(path / "schema.json", "w") as f:
+            f.write(self.serialize())
+
+        member_schemas = self.member_schemas()
+        for key, lf in self.to_dict().items():
+            member_schemas[key].sink_parquet(
+                lf,  # type: ignore
+                path / f"{key}.parquet",
+                **kwargs,
+            )
```

Comment:
did you consider dispatching to an internal `_to_parquet(directory: Path, sink: bool, **kwargs)` here to unify the sink and write methods, since they are mostly identical code?